compiler/sourcemap.nim

```diff
@@ -0,0 +1,377 @@
+import os, strformat, strutils, sequtils, tables, sets, ropes, json, algorithm
+
+type
+  SourceNode* = ref object
+    line*:      int
+    column*:    int
+    source*:    string
+    name*:      string
+    children*:  seq[Child]
+
+  C* {.pure.} = enum SourceNode, String
```

Comment:
This is a bit ridiculous. You want a ``pure`` enum and then a short notation? So use instead

```nim
type C* = enum cSourceNode, cString
```

Reply:
ok